### ISSUE TITLE
Question: support for multiple configurations?

### ISSUE THREAD

**Issue by snakescott** [pre-commit/pre-commit#432]
I'm interested in experimenting with different sets of hooks, having different hooks run on my build system versus locally, etc. 

Straightforward ways this might be possible:
- ``` pre-commit run --yaml $PATH --all-files```
- ```PRE_COMMIT_YAML_PATH=$PATH git commit -m "example"```

A more advanced option would be something like adding a ```groups``` yaml key as a sibling of  ```id``` e.g.

```yaml
-   repo: git://github.com/pre-commit/pre-commit-hooks
    sha: v0.4.2
    hooks:
    -   id: trailing-whitespace
        groups: [basic, advanced]
    -   id: flake8
        groups: [advanced]
```

You could then have ```pre-commit run --groups ...``` and perhaps ```pre-commit set-commit-groups ...```

I don't see anything like this discussed in the docs. Are there advanced/undocumented features that might support this? Is this functionality inline with pre-commit's project goals, or too complicated/promotes a workflow the project doesn't agree with?

**Comment by asottile** [comment]
So I can think of two ways to accomplish this in the current toolset, they're not the greatest but maybe they get you going in the right direction?

The `SKIP` environment variable allows you to temporarily disable hooks, so if I didn't want to run `autopep8` or `flake8` on an invocation I could run `SKIP=flake8,autopep8 pre-commit`. http://pre-commit.com/#temporarily-disabling-hooks

pre-commit can run individual hooks at a time by passing them on the commandline: `pre-commit run trailing-whitespace`.  You could do something where you maybe `echo 'trailing-whitespace flake8' | xargs -n1 pre-commit run`? (briefly mentioned here: http://pre-commit.com/#usage)

A tangentially related feature: you can already configure hooks to run at different stages (currently only pre-commit / pre-push) via http://pre-commit.com/#confining-hooks-to-run-at-certain-stages

As for implementing I'm a little concerned for a couple reasons:
- you can _kinda_ do it without adding additional tooling (yeah it's not the best, but it is possible)
- I'm not sure if the feature will get much use via the main usage of the tool: as a pre-commit hook (I'm not even sure how it would work with that, perhaps writing the configuration to  the .git/hooks/pre-commit executable, but then it's not idempotent any more and that complicates as well).
- at least for our usage we're running every hook every time on every commit

**Comment by snakescott** [comment]
Makes sense!

I think my underlying desire is to standardize my use of these kind of tools (autopep8, flake8, etc) on pre-commit. Adding the kind of flexibility I'm discussing also makes it easier to develop and deploy more complex workflows. For example I might want to experiment with using mypy through pre-commit (including on my build system, etc) without subjecting my team to a buggy hook-in-progress.

How do you feel about ```PRE_COMMIT_YAML_PATH```? Would you accept a PR for that? Seems like it would involve the least change. You could still have the same semantics (no unstaged changes to config file, etc), it would just un-hardcode the path. Should be completely backwards compatible as well?

**Comment by asottile** [comment]
Yeah I think adding that is pretty reasonable.   We're looking to move away from yaml soon to fix some stuff, let's go with `-c` `--config` as a cli parameter.  I think you'll want to augment each of these commands (`run`, `install`, `autoupdate`).

**Comment by snakescott** [comment]
Okay, I will hopefully have time to work on this next week! Thanks.

**Comment by asottile** [comment]
Sounds great!  Looking forward to it :)

**Comment by asottile** [comment]
This fix is in the (just released) 0.9.4 release.
